### PR TITLE
Implement error state for select

### PR DESCRIPTION
**Pull request by m-akinc**
# Pull Request

## 🤨 Rationale

TS/JS part of #632. Angular and Blazor updates will be in a follow-up PR. 

Nimble select should have invalid state styling and support for error-text attribute. See https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/6ec70d21-9a59-40cd-a8f4-45cfeed9e01e/specs/

## 👩‍💻 Implementation

- Added exclamation error icon to the end slot of the select
- Set the `order` CSS property of the selection region, error icon, and down arrow so that they are laid out in that order
- The combobox and select have shared styling that lives in patterns/dropdown/styles.ts. The combobox already supported invalid styling, so I moved that into the shared styles.ts.
- Added `error-text` and `error-visible` attributes to Select
- Updated Storybook stories for select

## 🧪 Testing

Tested in Storybook

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/select/tests/select-matrix.stories.ts [hunk 1]
```diff
@@ -23,9 +23,9 @@ import {
...
 /* array of state name, invalidClass, errorText */
 const selectInvalidStates = [
-    ['', ''],
-    ['invalid', 'This is not valid.'],
-    ['invalid', '']
+    [false, ''],
+    [true, 'This is not valid.'],
+    [true, '']
 ] as const;
 type SelectInvalidState = typeof selectInvalidStates[number];
```
Comment (rajsite) [inline packages/nimble-components/src/select/tests/select-matrix.stories.ts:31]: We now have some shared errorStates that are used by the other components using the ErrorPattern that we should leverage: https://github.com/ni/nimble/blob/2c07a38eec88d0e88977f407bdef98ada6ca8ee7/packages/nimble-components/src/utilities/tests/states.ts#L29-L34

File: packages/nimble-components/src/patterns/dropdown/styles.ts [hunk 2]
```diff
@@ -110,6 +111,14 @@ export const styles = css`
         border-color: rgba(${borderRgbPartialColor}, 0.1);
     }
 
+    :host([error-visible]) .control,
+    :host([error-visible][open]) .control,
+    :host([error-visible][disabled]) .control,
+    :host([error-visible]):before,
+    :host([error-visible]):after {
+        border-bottom-color: ${failColor};
+    }
```
Comment (rajsite) [inline packages/nimble-components/src/patterns/dropdown/styles.ts:120]: Please organize the selectors so they are grouped by the selector target (the host:before and host:after selectors should move to the right spots): https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/css-guidelines.md#group-selectors-by-target-and-document-order

File: packages/nimble-components/src/select/styles.ts [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+import { borderWidth } from '../theme-provider/design-tokens';
+import { appearanceBehavior } from '../utilities/style/appearance';
+import { DropdownAppearance } from './types';
 
 export const styles = css`
     ${dropdownStyles}
-`;
+    ${errorStyles}
+
+    [part='selected-value'] {
```
Comment (rajsite) [inline packages/nimble-components/src/select/styles.ts:12]: Can you add a comment for a similar disclaimer like the following but updated for this use-case:
https://github.com/ni/nimble/blob/9023d71ddb4740a624e5b9eac0114a0c91fbca78/packages/nimble-components/src/number-field/styles.ts#L150-L153

File: change/@ni-nimble-components-8dfe3583-abfb-4385-a130-bf75e17f7ed5.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (rajsite) [inline change/@ni-nimble-components-8dfe3583-abfb-4385-a130-bf75e17f7ed5.json:1]: Didn't see it in the PR description so just to make sure, will there be a follow-up with Angular and Blazor changes?

**Review by rajsite — APPROVED** [review]

File: change/@ni-nimble-components-8dfe3583-abfb-4385-a130-bf75e17f7ed5.json (hunk 4 above)
Comment (jattasNI) [reply change/@ni-nimble-components-8dfe3583-abfb-4385-a130-bf75e17f7ed5.json:1]: Could you update the PR description to mention #632 (and mark it as fixed if you add Angular and Blazor to this PR)?

File: packages/nimble-components/src/select/styles.ts [hunk 5]
```diff
@@ -1,6 +1,37 @@
...
 import { styles as dropdownStyles } from '../patterns/dropdown/styles';
+import { styles as errorStyles } from '../patterns/error/styles';
+import { borderWidth } from '../theme-provider/design-tokens';
+import { appearanceBehavior } from '../utilities/style/appearance';
+import { DropdownAppearance } from './types';
 
 export const styles = css`
     ${dropdownStyles}
-`;
+    ${errorStyles}
```
Comment (jattasNI) [inline packages/nimble-components/src/select/styles.ts:10]: I'm seeing a double underline in storybook when I turn on the error state and then tab focus into the component.
image

I tried this with a few other inputs and didn't see the double underline so it seems wrong to me, though I didn't check the XD spec.

**Review by jattasNI — APPROVED** [review]

File: packages/nimble-components/src/select/styles.ts (hunk 5 above)
Comment (m-akinc) [reply packages/nimble-components/src/select/styles.ts:10]: This is by design. Inputs that immediately get the cursor on focus (e.g. text field, numeric field, combo box) don't get this additional focus indicator. However, controls that don't show a blinking cursor on focus (e.g. tabs, button, checkbox) get a double underline/outline. This is not a change made in this PR (except for turning the double underline red when in the invalid state).

File: packages/nimble-components/src/patterns/dropdown/styles.ts (hunk 2 above)
Comment (m-akinc) [reply packages/nimble-components/src/patterns/dropdown/styles.ts:120]: Done

File: packages/nimble-components/src/select/styles.ts (hunk 3 above)
Comment (m-akinc) [reply packages/nimble-components/src/select/styles.ts:12]: Done

File: change/@ni-nimble-components-8dfe3583-abfb-4385-a130-bf75e17f7ed5.json (hunk 4 above)
Comment (m-akinc) [reply change/@ni-nimble-components-8dfe3583-abfb-4385-a130-bf75e17f7ed5.json:1]: Yes, I'll do those in a follow-on PR. Updated the PR description.

**Pull request merged**